lib/Standalone/MapConv2DNhwcHwcfToMatmul.cpp

```diff
@@ -109,24 +109,53 @@ static bool checkMappingToMatmul(linalg::LinalgOp linalgOp) {
   return match;
 }
 
-static bool preConditionsMapConvToGemm(linalg::LinalgOp linalgOp) {
+// Check if the operation looks like a convolution.
+static bool isConvLike(linalg::LinalgOp linalgOp) {
   if (linalgOp.getNumInputs() != 2)
     return false;
   if (linalgOp.getNumOutputs() != 1)
     return false;
-  if (linalgOp.hasBufferSemantics())
+  if (llvm::any_of(linalgOp.getInputAndOutputOperands(),
+                   [](OpOperand *operand) {
+                     return !operand->get().getType().isa<RankedTensorType>();
+                   }))
     return false;
-  return checkMappingToMatmul(linalgOp);
+  return tpp::hasMatmulBody(linalgOp);
+}
+
+static bool isInBound(unsigned filterRank, unsigned rPos, unsigned sPos) {
+  return !(rPos >= filterRank || sPos >= filterRank);
+    return false;
+  return true;
```

Comment:
Sorry, my bad, I should have selected all lines:
```suggestion
  return !(rPos >= filterRank || sPos >= filterRank);
```